src/commands/env/log.ts

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as Stream from 'stream';
+import * as util from 'util';
+import herokuColor from '@heroku-cli/color';
+import axios from 'axios';
+import { cli } from 'cli-ux';
+import * as Heroku from '@heroku-cli/schema';
+import EventSource = require('@heroku/eventsource');
+import { Messages } from '@salesforce/core';
+import { Errors, Flags } from '@oclif/core';
+import { FunctionsFlagBuilder } from '../../lib/flags';
+import Command from '../../lib/base';
+import { resolveAppNameForEnvironment } from '../../lib/utils';
+
+type EventSourceError = Error & {
+  status?: number;
+  message?: string;
+};
+
+// this splits a stream into lines
+function lineTransform() {
```

Comment:
It looks like a lot of these functions were copied over unchanged from the `tail` command (`lineTransform`, `eventSourceStream`, `readLogs`, maybe some others). Can we extract those to share between the two commands?